### PR TITLE
Sanitize country inputs

### PR DESCRIPTION
**Pull request by iriberri**
Fixes https://github.com/CartoDB/data-services/issues/194
Fixes https://github.com/CartoDB/cartodb/issues/3392

This PR sanitizes country inputs for admin1, namedplaces and postalcodes, both in the source SQL queries and in the cdb_geocoder extension.

## Manual tests run over this branch
**Note**: Automatic tests pass too.

**TEST 1:**
  * `SELECT (geocode_postalcode_points(Array['03204', '28010'], Array['Spain ', 'Spain'])).*;`

  old:
  ````
   q   |   c    | iso3 |                        geom                        | success 
-------+--------+------+----------------------------------------------------+---------
 03204 | Spain  |      |                                                    | f
 28010 | Spain  | ESP  | 0101000020E6100000575BB1BFEC9E0DC05A643BDF4F354440 | t
(2 rows)
  ````
  new:
  ````
   q   |   c    | iso3 |                        geom                        | success 
-------+--------+------+----------------------------------------------------+---------
 03204 | Spain  | ESP  | 0101000020E61000001D386744696FE6BF6F8104C58F214340 | t
 28010 | Spain  | ESP  | 0101000020E6100000575BB1BFEC9E0DC05A643BDF4F354440 | t
(2 rows)
  ````

  * `SELECT (geocode_postalcode_points(Array['03204', '28010'], 'Spain ')).*;`

  old:
  ````
   q   | geom | success 
-------+------+---------
 03204 |      | f
 28010 |      | f
(2 rows)
  ````
  new:
  ````
   q   |                        geom                        | success 
-------+----------------------------------------------------+---------
 03204 | 0101000020E61000001D386744696FE6BF6F8104C58F214340 | t
 28010 | 0101000020E6100000575BB1BFEC9E0DC05A643BDF4F354440 | t
(2 rows)
  ````

**TEST 2:**
  * `SELECT (geocode_postalcode_polygons(Array['11211', '11211'], Array['USA  ', 'United States  of America'])).*;`
  old:
  ````
   q   |             c             | success 
-------+---------------------------+---------
 11211 | USA                       | t
 11211 | United States  of America | t
(2 rows)
  ````
  new:
  ````
   q   |             c             | success 
-------+---------------------------+---------
 11211 | USA                       | t
 11211 | United States  of America | t
(2 rows)
  ````

  **Note**: This approach was correctly sanitized from the beginning: https://github.com/CartoDB/data-services/blob/a949e25dd64b400397f97c816ebdecaeed282e87/geocoder/postal-codes/sql/geocoder.sql#L18

  * `SELECT (geocode_postalcode_polygons(Array['11211'], ' USA')).q,  (geocode_postalcode_polygons(Array['11211'], ' USA')).success;`
  old:
  ````
   q   | success 
-------+---------
 11211 | f
(1 row)
  ````
  new:
  ````
   q   | success 
-------+---------
 11211 | t
(1 row)
  ````
**TEST 3:**
  * `SELECT (geocode_admin1_polygons(Array['Alicante', 'Barcelona '], 'Spain  ')).q,  (geocode_admin1_polygons(Array['Alicante', 'Barcelona '], 'Spain  ')).success;`
  old:
  ````
     q      | success 
------------+---------
 Alicante   | f
 Barcelona  | f
(2 rows)
  ````
  new:
  ````
     q      | success 
------------+---------
 Alicante   | t
 Barcelona  | t
(2 rows)


  ````

  * `SELECT (geocode_admin1_polygons(Array['Alicante', 'Barcelona '], Array['Spain', 'Spain  '])).q,  (geocode_admin1_polygons(Array['Alicante', 'Barcelona '], Array['Spain', 'Spain  '])).c,  (geocode_admin1_polygons(Array['Alicante', 'Barcelona '], Array['Spain', 'Spain  '])).success;`
  old:
  ````
     q      |    c    | success 
------------+---------+---------
 Alicante   | Spain   | t
 Barcelona  | Spain   | f
(2 rows)


  ````
  new:
  ````
     q      |    c    | success 
------------+---------+---------
 Alicante   | Spain   | t
 Barcelona  | Spain   | t
(2 rows)
  ````
**TEST 4:**
  * `SELECT (geocode_namedplace(Array['Elx'], Array['Spain '])).*;`
  old:
  ````
  q  |   c    | geom | success 
-----+--------+------+---------
 Elx | Spain  |      | f
(1 row)

  ````
  new:
  ````
  q  |   c    |                        geom                        | success 
-----+--------+----------------------------------------------------+---------
 Elx | Spain  | 0101000020E6100000739D465A2A6FE6BF28D53E1D8F214340 | t
(1 row)
  ````

  * `SELECT (geocode_namedplace(Array['Elx'], 'Spain ')).*;`
  old:
  ````
  q  |   c    | geom | success 
-----+--------+------+---------
 Elx | Spain  |      | f
(1 row)
  ````
  new:
  ````
  q  |   c    |                        geom                        | success 
-----+--------+----------------------------------------------------+---------
 Elx | Spain  | 0101000020E6100000739D465A2A6FE6BF28D53E1D8F214340 | t
(1 row)
  ````

### CODE REVIEW

**Comment by iriberri** [comment]
cc @rafatower

File: geocoder/admin1/sql/geocoder.sql [hunk 1]
```diff
@@ -1,11 +1,9 @@
 --Text array, country array --
-
-
 --- Function ---
 
 CREATE OR REPLACE FUNCTION geocode_admin_country_v1(names text[], country text[])
 RETURNS SETOF geocode_admin_country_v1 AS $$
-DECLARE 
+DECLARE
```
Comment (rafatower) [inline geocoder/admin1/sql/geocoder.sql]: We should deprecate this file

Comment (rafatower) [inline geocoder/admin1/sql/geocoder.sql]: even better: delete it

File: geocoder/extension/sql/0.0.1/50_postalcodes.sql [hunk 2]
```diff
@@ -199,21 +199,21 @@ CREATE FUNCTION geocode_postalcode_points(code text[], inputcountries text[]) RE
     SELECT
       q, c, iso3, geom, CASE WHEN geom IS NULL THEN FALSE ELSE TRUE END AS success
     FROM (
-      SELECT 
+      SELECT
         q, c, (SELECT iso3 FROM country_decoder WHERE
-                lower(d.c) = ANY (synonyms) LIMIT 1) iso3, (
+                lower(regexp_replace(d.c, '[^a-zA-Z\u00C0-\u00ff]+', '', 'g'))::text = ANY (synonyms) LIMIT 1) iso3, (
```
Comment (rafatower) [inline geocoder/extension/sql/0.0.1/50_postalcodes.sql]: how about encapsulating this regexp into a function?

Comment (iriberri) [inline geocoder/extension/sql/0.0.1/50_postalcodes.sql]: Sounds quite good. This way we can isolate it and make changes easier. On it.

File: geocoder/admin1/sql/geocoder.sql (hunk 1 above)
Comment (iriberri) [inline geocoder/admin1/sql/geocoder.sql]: https://github.com/CartoDB/data-services/pull/196

**Comment by iriberri** [comment]
I think this is good now. I encapsulated the regexp in a function and I got the same results in the manual tests. Automated tests pass too.

**Comment by rafatower** [comment]
:+1:

**Comment by rafatower** [comment]
now you can tag the third release candidate after merging :D

**Pull request merged**